plugwise/helper.py

```diff
@@ -459,9 +477,16 @@ def _appliance_info_finder(self, appliance: etree, appl: Munch) -> Munch:
         if appl.pwclass == "gateway":
             self.gateway_id = appliance.attrib["id"]
             appl.fw = self.smile_version[0]
+            appl.mac = self.smile_mac_address
             appl.model = appl.name = self.smile_name
             appl.v_name = "Plugwise B.V."
 
+            # Adam: check for ZigBee mac address
+            if self.smile_name == "Adam" and found := self._domain_objects.find(mac_locator
```

Comment:
My bad in the suggestion 🙏 

```suggestion
            if self.smile_name == "Adam" and found := self._domain_objects.find(
```

Reply:
This will work when you put () around found := ...
So
```
            if self.smile_name == "Adam" and (found := self._domain_objects.find(
                    ".//protocols/zig_bee_coordinator/mac_address"
                   )
            ):
```